notation/verifier.go

```diff
@@ -0,0 +1,159 @@
+	return v.name
+}
+
+// Type returns the type of the verifier which is always `notation`.
+func (v *Verifier) Type() string {
+	return notationVerifierType
+}
+
+// Verifiable returns true if the artifact is a Notation signature.
+func (v *Verifier) Verifiable(artifact ocispec.Descriptor) bool {
+	return artifact.ArtifactType == notationSignatureArtifactType && artifact.MediaType == ocispec.MediaTypeImageManifest
+}
+
+// Verify verifies the Notation signature.
+func (v *Verifier) Verify(ctx context.Context, opts *ratify.VerifyOptions) (*ratify.VerificationResult, error) {
+	subjectRef, err := registry.ParseReference(opts.Subject)
+	if err != nil {
+		return nil, fmt.Errorf("failed to parse subject reference: %w", err)
+	}
+
+	signatureDesc, err := v.getSignatureBlobDesc(ctx, opts.Store, subjectRef, opts.SubjectDescriptor)
+	if err != nil {
+		return nil, fmt.Errorf("failed to get signature blob descriptor: %w", err)
+	}
+
+	signatureBlob, err := opts.Store.FetchBlobContent(ctx, subjectRef.Repository, signatureDesc)
+	if err != nil {
+		return nil, fmt.Errorf("failed to fetch signature blob: %w", err)
+	}
+
+	result := &ratify.VerificationResult{
+		Verifier: v,
+	}
+	if outcome, err := v.verifySignature(ctx, opts.Subject, opts.SubjectDescriptor, signatureBlob, signatureDesc.MediaType); err != nil {
+		result.Err = err
+	} else {
+		cert := outcome.EnvelopeContent.SignerInfo.CertificateChain[0]
+		result.Detail = map[string]string{
+			"Issuer": cert.Issuer.String(),
+			"SN":     cert.Subject.String(),
+		}
+		result.Description = "Notation signature verification succeeded"
+	}
+
+	return result, nil
+}
+
+func (v *Verifier) getSignatureBlobDesc(ctx context.Context, store ratify.Store, artifactRef registry.Reference, artifactDesc ocispec.Descriptor) (ocispec.Descriptor, error) {
+	if artifactDesc.Size > maxManifestSizeLimit {
+		return ocispec.Descriptor{}, fmt.Errorf("signature manifest too large: %d bytes", artifactDesc.Size)
+	}
+
+	manifest, err := store.FetchImageManifest(ctx, artifactRef.Registry+"/"+artifactRef.Repository, artifactDesc)
+	if err != nil {
+		return ocispec.Descriptor{}, fmt.Errorf("failed to fetch image manifest for artifact: %w", err)
+	}
+
+	if len(manifest.Layers) != 1 {
+		return ocispec.Descriptor{}, fmt.Errorf("notation signature manifest requries exactly one signature envelope blob, got %d", len(manifest.Layers))
+	}
+
+	signatureDesc := manifest.Layers[0]
+	if signatureDesc.Size > maxBlobSizeLimit {
+		return ocispec.Descriptor{}, fmt.Errorf("signature blob too large: %d bytes", signatureDesc.Size)
+	}
+
+	return signatureDesc, nil
+}
+
+func (v *Verifier) verifySignature(ctx context.Context, subject string, subjectDesc ocispec.Descriptor, signature []byte, signatureMediaType string) (*notation.VerificationOutcome, error) {
+	opts := notation.VerifierVerifyOptions{
+		SignatureMediaType: signatureMediaType,
+		ArtifactReference:  subject,
+	}
+	return v.sigVerifier.Verify(ctx, subjectDesc, signature, opts)
+}
```

Comment:
nit: It seems not necessary to refactor this out.

Reply:
removed